### PR TITLE
Tag for release is correctly tagged to branch x.y.

### LINKED ISSUE #2124: releases are created with tag on latest commit on master instead of the passed commit hash
https://github.com/Mailu/Mailu/blob/3453d12ccbc2624b45178ddc1332c4e650b2c9ce/.github/workflows/CI.yml#L658
```
      - name: Tag and create release
        if: ${{ env.BRANCH != 'testing' && env.BRANCH != 'staging' && env.BRANCH != 'master' && env.PINNED_MAILU_VERSION != '' }}
        uses: ncipollo/release-action@v1
        with:
          bodyFile: "RELEASE_TEMPLATE.md"
          commit: ${{ env.GITHUB_SHA }}
          tag: ${{ env.PINNED_MAILU_VERSION }}
          token: ${{ secrets.GITHUB_TOKEN }}
```

Even though the action for creating the release receives the correct commit hash ( see env.GITHUB_SHA ), it tags the release/tag to the latest commit on the master branch. The tags should have been set to the following commits:
1.9.1: https://github.com/Mailu/Mailu/commit/89227e566524f24d0214ee0ff1a93f860b86d432
1.9.2: https://github.com/Mailu/Mailu/commit/bbca4f876ae4b97e7fdd918214f01b03ff2f4b32

The action documentation shows that we could also use a branch name instead of the commit hash. See https://github.com/ncipollo/release-action#notes

`
If the tag of the release you are creating does not yet exist, you should set both the tag and commit action inputs. commit can point to a commit hash or a branch name (ex - main).
`

We could try changing `commit: ${{ env.GITHUB_SHA }}` to `commit: ${{ env.BRANCH }}`. I hope the action "ncipollo/release-action@v1" works properly if we pass the branch (1.9) instead of the commit hash.

```
      - name: Tag and create release
        if: ${{ env.BRANCH != 'testing' && env.BRANCH != 'staging' && env.BRANCH != 'master' && env.PINNED_MAILU_VERSION != '' }}
        uses: ncipollo/release-action@v1
        with:
          bodyFile: "RELEASE_TEMPLATE.md"
          commit: ${{ env.BRANCH  }}
          tag: ${{ env.PINNED_MAILU_VERSION }}
          token: ${{ secrets.GITHUB_TOKEN }}
```

### PR DESCRIPTION
**Pull request by Diman0**
## What type of PR?

Bug-fix

## What does this PR do?
CI/CD had two issues which are addressed by this PR.
- calculating new x.y.z failed for x.y.10+
- tag was not created for branch x.y, but master. This resulted in the release pointing to the wrong commit.

backport is required to make this workflow available for 1.9.

### Related issue(s)
- closes #2124 

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -649,13 +649,19 @@ jobs:
...
         shell: bash
         run: |
-          version=$( git tag --list "${{ env.MAILU_VERSION }}.*" | tail -1  );root_version=${version%.*};patch_version=${version##*.};if [ "$patch_version" == "" ]; then pinned_version=${{ env.MAILU_VERSION }}.0; else pinned_version=$root_version.$(expr $patch_version + 1); fi;echo "PINNED_MAILU_VERSION=$pinned_version" >> $GITHUB_ENV
-      - name: Tag and create release
+          version=$( git tag --sort=version:refname --list "${{ env.MAILU_VERSION }}.*" | tail -1  );root_version=${version%.*};patch_version=${version##*.};if [ "$patch_version" == "" ]; then pinned_version=${{ env.MAILU_VERSION }}.0; else pinned_version=$root_version.$(expr $patch_version + 1); fi;echo "PINNED_MAILU_VERSION=$pinned_version" >> $GITHUB_ENV
+      - name: Create tag for branch x.y.
+        if: ${{ env.BRANCH != 'testing' && env.BRANCH != 'staging' && env.BRANCH != 'master' && env.PINNED_MAILU_VERSION != '' }}
+        shell: bash     
+        run: |
+          echo git tag ${{ env.PINNED_MAILU_VERSION }} $(/usr/bin/git log -1 --format='%H')
```
Comment (ghostwheel42) [inline .github/workflows/CI.yml:657]: Maybe you can use `git rev-parse HEAD` here?

**Review by ghostwheel42 — CHANGES_REQUESTED** [review]
I can't test this right now, but it lgtm.
Maybe use rev-parse instead of log?

File: .github/workflows/CI.yml (hunk 1 above)
Comment (Diman0) [reply .github/workflows/CI.yml:657]: I'm not very familiar with all git commands. After reading the help of git rev-parse and git rev-list I agree. Via command line it works as expected.
I prefer `$(/usr/bin/git rev-parse HEAD) over $(/usr/bin/git log -1 --format='%H')

Tonight I will change this and test that it still works as expected.
After that I will provide you access to my fork where you can test the workflow. I will make it easy for you to test it.

**Comment by Diman0** [comment]
I added you to the forked project I use for testing the CI/CD workflow. We used branch 1.10 for testing. To correctly test the workflow and releases for x.y and x.y.z:

1. Clone project https://github.com/Diman0/Mailu_Fork
2. On branch master create a dummy commit. E.g. create a new file and commit it and push it to the repository.
3. Switch to branch 1.10.
4. Cherry-pick the commit from step 2. Push this to the repository.
Once the workflow for branch 1.10 finished, double check in the release that the tag points to the commit of branch 1.10 and not master.

Branch

**Review by nextgens — APPROVED** [review]

**Pull request merged**